lib/matplotlib/ticker.py

```diff
@@ -2792,33 +2792,46 @@ def ideal_ticks(x):
         return MaxNLocator.tick_values(self, vmin, vmax)
 
     def nonsingular(self, vmin, vmax):
-        initial_range = (1e-7, 1 - 1e-7)
-        if not np.isfinite(vmin) or not np.isfinite(vmax):
-            return initial_range  # no data plotted yet
-
+        standard_minpos = 1e-7
+        initial_range = (standard_minpos, 1 - standard_minpos)
+        swap_vlims = False
         if vmin > vmax:
+            swap_vlims = True
             vmin, vmax = vmax, vmin
-
-        # what to do if a window beyond ]0, 1[ is chosen
-        if self.axis is not None:
-            minpos = self.axis.get_minpos()
-            if not np.isfinite(minpos):
-                return initial_range  # again, no data plotted
+        if not np.isfinite(vmin) or not np.isfinite(vmax):
+            vmin, vmax = initial_range  # Initial range, no data plotted yet.
+        elif vmax <= 0:
```

Comment:
perhaps group both into `if not 0 < vmin and vmax < 1: warn("Data has no values greater than 0 and less than 1, ...")`?

Reply:
it should be `not (0 < vmax and vmin < 1)` or `vmax <= 0 or vmin >= 1`. I prefer the second one. Except the operator priority, you invert `vmin` and `vmax`. This test is not intuitive to compare `vmax` to `0` and `vmin` to `1`, I have added a comment to understand that.

Also I change the sentence, which is clearer (it is no value).

A keep this thread open to check the wording.

Reply:
Sorry, I meant `not (0 < vmin <= vmax < 1)`.  Anyways the current wording and approach are fine too.

Reply:
Perfect. Thanks. I mark this thread as resolved.